packages/blocks/image/src/components/Media.tsx

```diff
@@ -266,11 +268,19 @@ export const Media: BlockComponent<
 
             if (existingLinkGroup) {
               await deleteLinks(
-                existingLinkGroup.links.map((link) => ({
-                  sourceAccountId: accountId,
-                  sourceEntityId: link.sourceEntityId,
-                  linkId: link.linkId,
-                })),
+                existingLinkGroup.links
+                  .filter(
+                    (
+                      link,
+                    ): link is BlockProtocolLink & SingleTargetLinkFields => {
```

Comment:
You don't need to do anything here as its likely a lot of work to change at this point, but this is kinda strange.

Really `BlockProtocolLink` should be setup as a generic – like this

```ts
export type BlockProtocolLink<LinkFields extends SingleTargetLinkFields | AggregationTargetLinkFields>  = {
  sourceAccountId?: string | null;
  sourceEntityId: string;
  sourceEntityTypeId?: string | null;
  sourceEntityVersionId?: string | null;
  destinationAccountId?: string | null;
  index?: number | null;
  path: string;
} & LinkFields;
```

And then you can do `link is BlockProtocolLink<SingleTargetLinkFields>`

Reply:
We're going to split up single target links from aggregation links entirely, so this is all temporary - I agree it is a bit of a mess at the moment (no fault of Akash's).